lib/resources/etc_group.rb

```diff
@@ -85,7 +85,8 @@ def where(conditions = {})
       conditions.each do |k, v|
         idx = fields[k.to_sym]
         next if idx.nil?
-        res = res.select { |x| x[idx] == v.to_s }
```

Comment:
Rather than selectively converting the value to an integer only if the field is `gid`, since we intend on comparing all values as strings, we should probably ensure the value we're comparing to is a string too. I think we should just change this line to be:

```ruby
res = res.select { |x| x[idx].to_s == v.to_s }
```

... rather than only converting the `gid`. This has the same issue if someone is looking for groups that have no members... empty members is represented as `nil` but we convert the match value to a string, which will be `""` -- so we may as well convert everything.